api/src/routers/meals.js

```diff
@@ -0,0 +1,165 @@
+import asyncHandler from "./error_handler.js";
+
+const mealsRouter = express.Router();
+
+mealsRouter.get(
+  "/",
+  asyncHandler(async (req, res) => {
+    const result = await knex("Meal").orderBy("id");
+    res.json({ result });
+  })
+);
+
+const validateMeal = (data) => {
+  const errors = [];
+
+  if (
+    data.title !== undefined &&
+    (!data.title || typeof data.title !== "string" || data.title.trim() === "")
+  ) {
+    errors.push("Title is required and must be a non-empty string");
+  }
+
+  if (
+    data.description !== undefined &&
+    (!data.description ||
+      typeof data.description !== "string" ||
+      data.description.trim() === "")
+  ) {
+    errors.push("Description is required and must be a non-empty string.");
+  }
+
+  if (
+    data.location !== undefined &&
+    (!data.location ||
+      typeof data.location !== "string" ||
+      data.location.trim() === "")
+  ) {
+    errors.push("Location is required and must be a non-empty string.");
+  }
+
+  if (
+    data.meal_when !== undefined &&
+    (!data.meal_when || isNaN(Date.parse(data.meal_when)))
+  ) {
+    errors.push("Meal_when must be a valid date.");
+  }
+
+  if (data.max_reservations !== undefined) {
+    const maxReservations = Number(data.max_reservations);
+    if (!Number.isInteger(maxReservations) || maxReservations < 1) {
+      errors.push("Max reservations must be a positive integer.");
+    }
+  }
+
+  if (data.price !== undefined) {
+    const price = Number(data.price);
+    if (typeof price !== "number" || price <= 0) {
+      errors.push("Price must be a positive number.");
+    }
+  }
+
+  return errors;
+};
+
+mealsRouter.post(
+  "/",
+  asyncHandler(async (req, res) => {
+    const { title, description, location, meal_when, max_reservations, price } =
+      req.body;
+    const data = {
+      title,
+      description,
+      location,
+      meal_when,
+      max_reservations: Number(max_reservations),
+      price: Number(price),
+    };
+    const errors = validateMeal(data);
+
+    if (errors.length > 0) {
+      return res.status(400).json({ errors });
+    }
+    const newMeal = await knex("Meal").insert({
+      title,
+      description,
+      location,
+      meal_when,
+      max_reservations,
+      price,
+    });
+
+    res.status(201).json({ status: "success", "id of the new meal": newMeal });
+  })
+);
+
+mealsRouter.get(
+  "/:id",
+  asyncHandler(async (req, res) => {
+    const id = Number(req.params.id);
+    const meal = await knex("Meal").where({ id }).first();
+    if (!meal) {
+      res.status(404).send("Meal not found");
+    } else {
+      res.json(meal);
+    }
+  })
+);
+
+mealsRouter.put(
+  "/:id",
+  asyncHandler(async (req, res) => {
+    const id = Number(req.params.id);
+    const {
+      title,
+      description,
+      location,
+      meal_when,
+      max_reservations,
+      price,
+      created_date,
+    } = req.body;
+
+    const updateData = {};
```

Comment:
I generally recommend as a practice to name "active" elements such as functions using "active" names like `update` and other way around for passive, such as objects fx. `toUpdate` or `updated`. In this case, the object is collecting data to be updated, right? I would suggest `dataToUpdate` or something similar :)

Reply:
it does sound better! Thank you!